File: engine/src/main/java/de/hanno/hpengine/engine/component/BaseComponent.java
package de.hanno.hpengine.engine.component;

import de.hanno.hpengine.engine.entity.Entity;

import java.io.Serializable;

public abstract class BaseComponent implements Component, Serializable {
    private static final long serialVersionUID = -224913983270697337L;

    protected Entity entity;

    public String getIdentifier() { return this.getClass() + " " + System.currentTimeMillis(); }

    @Override
    public Entity getEntity() {
        return entity;
    }

}

File: engine/src/main/java/de/hanno/hpengine/engine/component/BaseComponent.kt
package de.hanno.hpengine.engine.component

import de.hanno.hpengine.engine.entity.Entity

import java.io.Serializable

abstract class BaseComponent(override val entity: Entity) : Component, Serializable {

    companion object {
        private const val serialVersionUID = -224913983270697337L
    }

}
